libtiledbsoma/src/soma/soma_context.h

```diff
@@ -47,10 +48,14 @@ class SOMAContext {
     //= public non-static
     //===================================================================
     SOMAContext()
-        : ctx_(std::make_shared<Context>(Config({}))){};
+        : ctx_(std::make_shared<Context>(Config({}))) {
+        create_thread_pool();
+    };
 
     SOMAContext(std::map<std::string, std::string> platform_config)
-        : ctx_(std::make_shared<Context>(Config(platform_config))){};
+        : ctx_(std::make_shared<Context>(Config(platform_config))) {
+        create_thread_pool();
+    };
 
```

Comment:
I don't think we should create the threadpool whenever we construct the `SOMAContext`. Only when we need to use it.

Reply:
@beroy - did this comment get addressed?  Unless I misunderstand the code, it it not lazy?